### PR TITLE
Create initial test suite for ec2_metadata_facts

### PR DESCRIPTION
**Pull request by jillr**
##### SUMMARY
Create an instance, run metadata on the remote instance, and teardown.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
tests/integration/targets/ec2_metadata_facts/

### CODE REVIEW

**Comment by jillr** [comment]
@tremble This actually ended up being less hacky than I thought it would. Thoughts?

**Comment by jillr** [comment]
Hrm. TODO: handle running teardown.yml even if test_initial_metadata.yml fails, ssh works on my machine but not shippable?

File: tests/integration/targets/ec2_metadata_facts/playbooks/test_initial_metadata.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (tremble) [inline tests/integration/targets/ec2_metadata_facts/playbooks/test_initial_metadata.yml:1]: We should probably add a wait_for_connection entry here.  The failure looks to be an auth failure which is likely the connection occurring before the user-data finishes running.

File: tests/integration/targets/ec2_metadata_facts/runme.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+set -eux
+export ANSIBLE_HOST_KEY_CHECKING=False
+export ANSIBLE_SSH_ARGS='-o UserKnownHostsFile=/dev/null'
+
+# create test resources and inventory
+ansible-playbook playbooks/setup.yml -c local "$@"
+
```
Comment (tremble) [inline tests/integration/targets/ec2_metadata_facts/runme.sh:9]:
```suggestion

# Destroy Environment
cleanup() {
    ansible-playbook playbooks/teardown.yml -i inventory -c local "$@"
}

trap "cleanup" EXIT

```

(stolen from connection_aws_ssm integration tests)

**Review by tremble — CHANGES_REQUESTED** [review]

**Review by tremble — APPROVED** [review]
LGTM

**Pull request merged**